lib/libutee/tee_api_slf_bridge_gp11.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * Short-lived functions
+ * Bridge to Global Platform Internal Core API 1.1
+ * These functions will be removed at some future major revision of
+ * this specification
+ */
+
+#include <tee_api.h>
+
+/* Based on GP TEE Internal API Specification Version 1.1 */
+/* #define SLF_BRIDGE_GP11_ENABLED */
+
+void slf_bridge_gp11_check_objectinfo_rc(TEE_ObjectInfo *objectInfo)
+{
+
+	if (objectInfo == TEE_HANDLE_NULL)
+		return;
+
+#ifdef SLF_BRIDGE_GP11_ENABLED
+	if (objectInfo->objectType == TEE_TYPE_CORRUPTED_OBJECT) {
```

Comment:
Why not do this directly in TEE_GetObjectInfo() instead? I don't think we break 1.0 compliance by having it there, by hiding it here it's just harder to follow the code.